eng/source-build-pre-release.yml

```diff
@@ -0,0 +1,146 @@
+trigger: none
+pr: none
+
+pool:
+  name: NetCore1ESPool-Svc-Internal
+  demands: ImageOverride -equals 1es-ubuntu-2004
+
+resources:
+  pipelines:
+  - pipeline: dotnet-staging-pipeline-resource
+    source: Stage-DotNet
+
+variables:
+- group: DotNet-MSRC-Storage
+- name: azDoOrg
+  value: 'https://dev.azure.com/dnceng/'
+- name: azDoProject
+  value: 'internal'
+- name: installerOfficialCiPipelineId
+  value: '286'
+- name: installerTarabllBuildCiPipelineId
+  value: '1011'
+- name: storageAccountName
+  value: 'dotnetclimsrc'
+- name: blobContainerName
+  value: 'source-build'
+- name: blobContainerUploadBaseFilePath
+  value: 'release'
+
+stages:
+
+- stage: GetBuildInfoStage
```

Comment:
Why a different stage for retrieving the build info?  My concern is that it will take at least one order order of magnitude longer to acquire a build agent than both stages take to run.

Reply:
I used a different stage because I imagine that in the future multiple stages could run using the variables set by this stage (e.g. a separate stage for updating VMR).

Do you think these should be jobs instaed of stages and just run everything at once? Or do you think running `GetBuildInfoJob` in every stage would be faster?

Reply:
I think these overall these steps aren't going to take much time and provisioning machines for each is going to be wasteful (e.g. plan for machine provisioning to take up to 20 minutes).  I would recommend utilizing as few stages/jobs as possible.

Reply:
In my latest update I reduced everything to one job and multiple stages, and pulled out "Get Build Info" to a template so that it can be used in multiple jobs/pipelines.